### ISSUE TITLE
filter_access_to inheritance and if_attribute

### ISSUE THREAD

**Issue by benlangfeld** [stffn/declarative_authorization#58]
I have models for User and Domain. User has_one domain and Domain has_many users.

I have an admin role and Domain::admins, and this all works fine as in this gist: http://gist.github.com/442490

I then have the following auth rule:

```
role :admin do
  has_permission_on :domains, :to => [:show, :manage] do
    if_attribute :admins => contains {user}
  end

  has_permission_on :users, :to => [:show, :manage] do
    if_permitted_to :manage, :domain
  end
end
```

where :manage includes :edit, :update, :destroy

So you would expect that the user with ID 2 would be able to edit only one of the domains shown in the gist. Unfortunately, this is not the case, and this user can edit both domains. It would appear that if_attribute is returning true indiscriminately.

Is this an issue with declarative_authorization or something funny in my code? How might I go about debugging further?

**Comment by benlangfeld** [comment]
Ok, so I figured out what was causing this. Using filter_resource_access in individual controllers rather than filter_access_to :all in the ApplicationController solves this instantly. I'm not sure if this is buried in the documentation anywhere, but i'll submit a pull request for a modification to the README.

**Comment by stffn** [comment]
Could this be related to :attribute_check => true?  Else, I'd say this might really be a bug

**Comment by benlangfeld** [comment]
:attribute_check sounds like it's definitely the issue. Better README patch submitted.

**Comment by stffn** [comment]
Doc update to stress :attribute_check
(Closed by 8a1e369847bb451d9f4e2c0f5eb67caaea4062cc, thanks benlangfeld)
